platform/linux-generic/odp_traffic_mngr.c

```diff
@@ -2250,6 +2254,16 @@ static void tm_send_pkt(tm_system_t *tm_system, uint32_t max_sends)
 		if (tm_system->egress.egress_kind == ODP_TM_EGRESS_PKT_IO) {
 			ret = odp_pktout_send(tm_system->pktout, &odp_pkt, 1);
 			if (odp_unlikely(ret != 1)) {
+				pktio_entry = get_pktio_entry(tm_system->pktout.pktio);
+				pkt_hdr = packet_hdr(odp_pkt);
+				if (odp_unlikely(_odp_pktio_tx_compl_enabled(pktio_entry) &&
+						 pkt_hdr->p.flags.tx_compl)) {
+					buf = odp_buffer_alloc(pktio_entry->s.tx_compl_pool);
+					if (buf != ODP_BUFFER_INVALID)
+						_odp_pktio_send_tx_compl_event(buf,
+									       pkt_hdr->user_ptr,
+									       pkt_hdr->dst_queue);
+				}
```

Comment:
Instead of using `_odp_pktio_send_tx_compl_event()`, `allocate_and_send_tx_compl_events()` could be used, which minimizes the amount of code required in the TM module.

Reply:
Done in v7.